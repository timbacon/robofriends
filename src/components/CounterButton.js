import React, {useState} from 'react';

const CounterButton = ({color}) => {
  const [count, setCount] = useState(0)

  return (
    <button
        id='counter'
        color={color}
        count={count}
        onClick={() => setCount(count => count + 1)}>
        Count: {count}
      </button>
  )
}

export default CounterButton