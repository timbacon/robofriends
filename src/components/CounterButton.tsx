import React, {useState} from 'react';

interface IProps {
  color?: string
}

const CounterButton = ({color}: IProps) => {
  const [count, setCount] = useState(0)

  return (
    <button
        id='counter'
        color={color}
        // count={count}
        onClick={() => setCount(count => count + 1)}>
        Count: {count}
      </button>
  )
}

export default CounterButton