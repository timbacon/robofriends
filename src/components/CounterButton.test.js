import { shallow } from 'enzyme';
import React from 'react';
import CounterButton from './CounterButton';

test('expect to render CounterButton component', () => {
    const mockColor = 'red'
    expect(shallow(<CounterButton color={mockColor}/>)).toMatchSnapshot();
})

test('should correctly increments the counter', () => {
    const mockColor = 'red'
    const wrapper = shallow(<CounterButton color={mockColor} />)

    wrapper.find('[id="counter"]').simulate('click');
    wrapper.find('[id="counter"]').simulate('click');
    expect(wrapper.props().count).toEqual(2);
})
