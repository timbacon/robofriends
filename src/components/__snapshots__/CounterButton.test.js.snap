// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CounterButton component 1`] = `
<button
  color="red"
  count={0}
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;
